### PR TITLE
Feature/1610.2 frontend parsing status

### LINKED ISSUE #1610: As a user, I want information about the acceptance of my data and a link to the error report for each version
**Description:**
_Provide a brief background and justification for this issue_
This is for Release 3 (post parsing submission history)
For the next iteration of these tables, we need to provide the users new columns: acceptance status and error report links.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] Users can see a graphic acceptance status per Figma spec
- [ ] Table retains USWDS compliance and form
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [x] Creation of new columns: "Acceptance Status"
- [x] Creation of icons for acceptance status
- [x] Populate acceptance status column from metadata of the respective filename
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

image

- [Figma mockup](https://www.figma.com/file/irgQPLTrajxCXNiYBTEnMV/TDP-Mockups-For-Feedback?node-id=6576%3A49872)
- [User story from research](https://hackmd.io/xaYQjwFmRpOIonSM1cMRNA#Validated-amp-Dev-Ready-Stories)

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_

### PR DESCRIPTION
**Pull request by jtimpe**
## Summary of Changes

Pull request closes #1610 
* implements the `DatafileSummary` model in the `/v1/data_files` api and on the frontend Submission history

## How to Test
```
cd tdrs-backend && docker-compose up
cd tdrs-frontend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
1. Navigate to the Data files page and submit one of the test files
1. Navigate to the Submission History tab and verify the Acceptance Status column matches expected
1. Repeat with other data files, recreating the possible error messages ("Accepted", "Accepted with Errors", and "Rejected")

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Users can see a graphic acceptance status per Figma spec
+ [x] Table retains USWDS compliance and form
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [x] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [x] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx [hunk 1]
```diff
@@ -11,6 +17,31 @@ import { getParseErrors } from '../../actions/createXLSReport'
 
 const formatDate = (dateStr) => new Date(dateStr).toLocaleString()
 
+const SubmissionSummaryStatusIcon = ({ status }) => {
+  let icon = null
+  let color = null
+
+  switch (status) {
+    case 'Accepted':
```
Comment (jtimpe) [inline tdrs-frontend/src/components/SubmissionHistory/SubmissionHistory.jsx:30]: missed pending

**Comment by raftmsohani** [comment]
When I uploaded the files, sometimes the status was not changing from Pending, but it is not in the scope for this ticket. Looks like the task was stuck. everythign else looks good to me

**Review by raftmsohani — COMMENTED** [review]
we are getting deploy failure but not related to this ticket, everything looks good to me

**Review by andrew-jameson — APPROVED** [review]

**Review by andrew-jameson — APPROVED** [review]
It feels so crazy to finally see this application coming to life. Love it!

The code is functional and fine with me besides that one "pass" unit test. I would like to have a conversation with @reitermb  on the utility of showing the end user the two different statuses "*with errors" that share the same icon/color. Possible source of confusion -- maybe we just show them the one status? I know it will be useful on the backend to have the distinct granularity there when we go into 1612 and possible other work later.  Otherwise approved!

Screenshot 2023-08-24 at 10 22 36 AM

File: tdrs-backend/tdpservice/parsers/test/test_models.py [hunk 2]
```diff
@@ -21,3 +21,8 @@ def test_parser_error_rpt_month_name(parser_error_instance):
     """Test that the parser error instance is created."""
     parser_error_instance.rpt_month_year = 202001
     assert parser_error_instance.rpt_month_name == "January"
+
+
+@pytest.mark.django_db
+def test_datafile_summary_status_approved():
+    pass
```
Comment (andrew-jameson) [inline tdrs-backend/tdpservice/parsers/test/test_models.py]: This was the test I meant. Vestigial or did you want to flesh this out?

Comment (jtimpe) [reply tdrs-backend/tdpservice/parsers/test/test_models.py]: vestigial, before i realize the case i needed _was_ covered, my code just wasn't triggering the failure properly 🤦 

good catch, thank you

**Review by raftmsohani — APPROVED** [review]
LGTM! Tested the function

File: tdrs-backend/tdpservice/parsers/migrations/0007_alter_datafilesummary_datafile.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('data_files', '0012_datafile_s3_versioning_id'),
+        ('parsers', '0006_auto_20230726_1448'),
```
Comment (andrew-jameson) [inline tdrs-backend/tdpservice/parsers/migrations/0007_alter_datafilesummary_datafile.py]:
```suggestion
        ('parsers', '0006_auto_20230810_1500'),
```
Looks like this was recreated at a different date, updating.

**Comment by reitermb** [comment]
@jtimpe Still looking to give this a quick pass when possible in a11y, is there another dependency that needs to be merged in to enable that? Still getting blank submission history & request failed w/ status code 500 errors on submission when I deploy it.

**Review by reitermb — APPROVED** [review]
Looking good on a11y 🚀 

Clear axe scan is attached and the table plays nicely with screenreaders as well! 

image

cc @ADPennington @ttran-hub

**Comment by reitermb** [comment]
Officially bumping into QASP a11y/dev review post sync with @ttran-hub.

cc @ADPennington

**Comment by ttran-hub** [comment]
> Looking good on a11y 🚀
> 
> Clear axe scan is attached and the table plays nicely with screenreaders as well!
> 
> image
> 
> cc @ADPennington @ttran-hub

a11y 🚀 cc @ADPennington

**Review by George-Hudson — APPROVED** [review]

**Review by ADPennington — APPROVED** [review]
thanks @jtimpe 🚀 

want to revisit the order of operations with the file validation soon and the status definitions, particularly "partially accepted" and "rejected"

**Pull request merged**